js/views/install_view.js

```diff
@@ -162,11 +163,29 @@
                     }
 
                     this.selectStep(Steps.PROGRESS_BAR);
-                    resolve(name);
+
+                    var finish = function() {
+                        resolve(name);
+                    };
+
+                    // Delete all data from database unless we're in the middle
+                    //   of a re-link, or we are finishing a light import. Without this,
+                    //   app restarts at certain times can cause weird things to happen,
+                    //   like data from a previous incomplete light import showing up
+                    //   after a new install.
+                    if (this.retainData) {
+                        return finish();
+                    }
+
+                    Whisper.Backup.clearDatabase().then(finish, function(error) {
+                        console.log(
+                          'confirmNumber: error clearing database',
+                          error && error.stack ? error.stack : error
+                        );
+                        finish();
+                    });
```

Comment:
Curious: What is the difference between using the `onReject` handler vs `then(onResolve).catch(error => …)`?

Reply:
With `then(onResolve).catch(errorHandler)`, `errorHandler` will be run after any error thrown from `onResolve`.

With `then(onResolve, errorHandler)`, any error thrown from `onResolve` will result in a rejected promise.